dotcom-rendering/src/components/Island.tsx

```diff
@@ -133,18 +134,30 @@ export const Island = ({
 	placeholderHeight,
 	rootMargin,
 	children,
-}: Props) => (
-	<gu-island
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access -- Type definitions on children are limited
-		name={children.type.name}
-		deferUntil={deferUntil}
-		props={JSON.stringify(children.props)}
-		clientOnly={clientOnly}
-		rootMargin={rootMargin}
-	>
-		{decideChildren(children, clientOnly, placeholderHeight)}
-	</gu-island>
-);
+}: Props) => {
+	/**
+	 * Where is this coming from?
+	 * Config value is set at high in the component tree within a React context
+	 * @see /dotcom-rendering/src/components/ArticlePage.tsx or look for a `<ConfigProvider />`
+	 *
+	 * This is here so that we can provide the config information to the hydrated, client-side rendered components
+	 */
+	const config = useConfig();
+
+	return (
+		<gu-island
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access -- Type definitions on children are limited
+			name={children.type.name}
+			deferUntil={deferUntil}
+			props={JSON.stringify(children.props)}
+			clientOnly={clientOnly}
+			rootMargin={rootMargin}
+			config={JSON.stringify(config)}
```

Comment:
Is this the exact same data serialised for each and every island?

Reply:
We'll probably end up repeating the same data a lot as it's the same config for each island. I've added a comment to `initHydration`, but I wonder if we could put config in one place and have islands look it up from there, instead of passing it as an attribute to every island element?

Reply:
You could serialise with `JSON.stringify` inside a single JSON script on the page?

Reply:
To @mxdvl's suggestion here's an [example](https://github.com/guardian/dotcom-rendering/blob/main/apps-rendering/src/server/page.tsx#L110-L112) of using the script tag from `apps-rendering`